### ISSUE TITLE
Add organism and strain to annotation tables

### ISSUE THREAD

**Issue by jseager7** [pombase/canto#1715]
(Follow-up from #1712)

The organism name and strain name is not displayed in any annotation table except in the case of metagenotypes. This is a problem when the gene names are identical across species:

![image](https://user-images.githubusercontent.com/11753634/49731684-18f64a00-fc74-11e8-9177-b340b63dbb9d.png)

@kimrutherford I've been trying to figure out the template hierarchy for annotation tables on the front page, this is my current understanding:

- `annotation_table_list`
  - `annotation_table`  
     Then the following templates are dynamically chosen based on a binding in the `annotationTableRow` directive:
    - `annotation_table_interaction_row.html`
    - `annotation_table_ontology_row.html`

I presume I'll have to add column headers to `annotation_table`, and add the cell values to the dynamic templates (`annotation_table_<type>_row.html`, but are there any other templates I should be aware of?

**Comment by kimrutherford** [comment]
> I presume I'll have to add column headers to annotation_table, and add the cell values to the dynamic
> templates

Yep, that's right.  The new column will need to be multi-organism mode only.

> but are there any other templates I should be aware of?

That's all for now.

Later we'll need to decide what to do about the interaction annotation types.  They might need extra columns too.  That will require a chat or two at our Monday Skype meetings.

**Comment by jseager7** [comment]
I've managed to add organism and strain columns to the pathogen / host phenotype annotations, and I've ensured that the columns don't appear in the metagenotype annotation rows (since the organism and the strain are already included in the _Pathogen genotype_ and _Host genotype_ colums).

Unfortunately, the values for organism and strain aren't retrieved in the case of wild-type annotations, like _GO molecular function_:

![1715_wt_annotation_no_organism](https://user-images.githubusercontent.com/37659591/50893901-e453ea80-13f9-11e9-9b9b-1f50f82d04d6.PNG)

Here's the fields I'm accessing to retrieve these values, which works for genotypes:

```html
  <td ng-if="multiOrganismMode && annotationType.feature_type !== 'metagenotype'">
    {{annotation.organism.scientific_name}}
  </td>
  <td ng-if="multiOrganismMode && annotationType.feature_type !== 'metagenotype'">
    {{annotation.strain_name}}
  </td>
```
I presume these values aren't included for wild-type annotations? Could they be included, or are there other fields I could use?

**Comment by ValWood** [comment]
We don't need the strain info for GO annotations.

**Comment by kimrutherford** [comment]
> We don't need the strain info for GO annotations.

Yep, we should hide the strain column for non-phenotype annotations.

**Comment by jseager7** [comment]
> hide the column unless it's a phenotype annotation.

I've made this change so that the strain column doesn't show in gene annotations.

- - -

Unfortunately I'm still stuck on the fact that the server doesn't seem to provide the organism as part of the data it returns. I've tried to trace where the Perl code gathers the data, and I think it's related to the `make_ontology_annotation` subroutine in `Utils.pm`, which returns this:

https://github.com/pombase/canto/blob/ea0e7f462ea3b7e0b30d09c4807468c031ad626e/lib/Canto/Curs/Utils.pm#L316-L347

This subroutine provides the taxon ID of the organism, but not the organism data itself, unlike `_make_genotype_details`, which provides an object (hash?) for the organism data:

https://github.com/pombase/canto/blob/ea0e7f462ea3b7e0b30d09c4807468c031ad626e/lib/Canto/Curs/Utils.pm#L107-L119

@kimrutherford Could this organism data be provided by `make_ontology_annotation` as well? Should it already be there?
